hw_diag/utilities/dashboard_registration.py

```diff
@@ -0,0 +1,106 @@
+import requests
+import json
+import logging
+from collections import namedtuple
+from urllib.parse import urlencode, urlunparse, urlparse
+
+from hw_diag.utilities.db import get_value, set_value
+from hw_diag.utilities.hardware import is_nebra_device
+from hw_diag.utilities.diagnostics import cached_diagnostics_data
+from hw_diag.constants import DIAG_JSON_KEYS
+
+THIRD_PARTY_MINER_REGISTRATION_URL = \
+    "https://dashboard.nebra.com/api/v0.1/register-third-party-hotspot"
+MINER_CLAIM_URL = "https://dashboard.nebra.com/devices/#qr-modal"
+
+log = logging.getLogger()
+log.setLevel(logging.DEBUG)
+
+
+def get_value_noexcept(key: str, default: str | None = None) -> str | None:
+    '''get value if exits otherwise default value'''
+    try:
+        return get_value(key)
+    except Exception as e:
+        logging.debug(f"exception while retrieving {key}: {e}, using default")
+        return default
+
+
+def set_value_noexcept(key: str, value: str) -> bool:
+    '''return true if value can be set in db without error'''
+    try:
+        set_value(key, value)
+    except Exception as e:
+        logging.error(f"failed to set value {e}")
+        return False
+
+
+def _prepare_registration_payload(diagnostics: dict) -> dict:
+    device_payload = {
+        "ethernet_mac_address": diagnostics[DIAG_JSON_KEYS.ETH_MAC_ADDRESS],
+        "wifi_mac_address":  diagnostics[DIAG_JSON_KEYS.WLAN_MAC_ADDRESS],
+        "frequency": diagnostics[DIAG_JSON_KEYS.FREQUENCY],
+        "balena_uuid": diagnostics[DIAG_JSON_KEYS.BALENA_UUID],
+        "onboarding_address": diagnostics[DIAG_JSON_KEYS.ONBOARDING_KEY],
+        "public_address": diagnostics[DIAG_JSON_KEYS.PUBLIC_KEY],
+        "serial_number": diagnostics[DIAG_JSON_KEYS.SERIAL_NUMBER],
+        "variant": diagnostics[DIAG_JSON_KEYS.VARIANT]
+    }
+    return device_payload
+
+
+def register_third_party_miner() -> None:
+    '''
+    tries to register a non nebra miner with dashboard
+    '''
+    try:
+        diagnostics = cached_diagnostics_data()
```

Comment:
@kashifpk Can we remove this check. I think we get "400, already registered for nebra miners".  I don't really know the background of why this is a seaprate api.

Reply:
@pritamghanghas Haven't looked at this code before but from what I see in the code, this should just be about caching local diagnostics data and not affect dashboard registration. Better keep it in place as it's probably meant to reduce actual diagnostic runs.